### ISSUE TITLE
Harness ends with DEADLYSIGNAL at start

### ISSUE THREAD

**Issue by AdvenamTacet** [trailofbits/ruzzy#20]
Often running a harness ends with:
```
AddressSanitizer: DEADLYSIGNAL
AddressSanitizer: DEADLYSIGNAL
AddressSanitizer: DEADLYSIGNAL
...
AddressSanitizer: DEADLYSIGNAL
AddressSanitizer: DEADLYSIGNAL
AddressSanitizer: DEADLYSIGNAL

```

In my experience, it always happens at the very beginning of running the harness.

While my setup is quite specific, I believe @mschwager observed the same behavior. (Am I right?)

**Comment by mschwager** [comment]
Yeah, I've seen this reliably a number of times. In my case, it would crash libFuzzer and the fuzzing process. This was from a while ago, so I don't immediately have the reproduction steps, but I did dig up these links that I was using to debug:

- https://stackoverflow.com/questions/77826203/addresssanitizer-randomly-throws-sigsegv-with-no-explanation
- https://stackoverflow.com/questions/76514542/inconsistent-output-from-addresssanitizer-intermittent-error-segv-on-unknown-ad
- https://groups.google.com/g/emscripten-discuss/c/QERnFWUT0_A
- https://github.com/google/sanitizers/issues/1724
- https://github.com/google/sanitizers/issues/1716

I did try turning off ASLR, but that didn't stop the issue from happening for me.
